File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.js
class RandomizedSet {
    constructor() {
        this.map = new Map()
        this.arr = []
    }

    insert(val) {
        if(this.map.has(val)) return false
        this.map.set(val, this.arr.length)
        this.arr.push(val)
        return true
    }
    remove(val) {
        if(this.map.has(val)) {
            const ind = this.map.get(val)
            const lastEl = this.arr[this.arr.length-1]
            this.map.delete(val)
            if(lastEl !== val) {
                this.arr[ind] = lastEl
                this.map.set(lastEl, ind)
            }
            this.arr.pop()
            return true
        }
        return false
    }
    getRandom() {
        const rand = parseInt(Math.random() * this.arr.length)
        return this.arr[rand]
    }
}

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.py
class RandomizedSet:

    def __init__(self):
        self.map = {}
        self.arr = []

    def insert(self, val: int) -> bool:
        if val in self.map:
            return False
        self.arr.append(val)
        idx = len(self.arr) - 1
        self.map[val] = idx
        return True

    def remove(self, val: int) -> bool:
        if val not in self.map:
            return False
        idx = self.map[val]
        last = self.arr[-1]
        self.arr[idx] = last
        self.arr.pop()
        self.map[last] = idx
        self.map.pop(val)
        return True

    def getRandom(self) -> int:
        rand = math.floor(random.random() * len(self.arr))
        return self.arr[rand]


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()